src/ButtonBase/TouchRipple.tsx

```diff
@@ -5,93 +5,90 @@ import * as classNames from 'classnames';
 import * as TransitionGroup from 'react-transition-group/TransitionGroup';
 import withStyles from '../styles/withStyles';
 import Ripple, { RippleClassKey, RippleProps } from './Ripple';
-import { Theme, WithStyles } from '../styles';
+import { StyleRulesCallback, Theme, WithStyles } from '../styles';
 
 const DURATION = 550;
 export const DELAY_RIPPLE = 80;
 
 export type TouchRippleClassKey = 'root' | RippleClassKey;
-export const styles = withStyles<TouchRippleClassKey>(
-  (theme: Theme) => ({
-    root: {
-      display: 'block',
-      position: 'absolute',
-      overflow: 'hidden',
-      borderRadius: 'inherit',
-      width: '100%',
-      height: '100%',
-      left: 0,
-      top: 0,
-      pointerEvents: 'none',
-      zIndex: 0,
-    },
-    wrapper: {
-      opacity: 1,
-    },
-    wrapperLeaving: {
-      opacity: 0,
-      animation: `mui-ripple-exit ${DURATION}ms ${theme.transitions.easing.easeInOut}`,
-    },
-    wrapperPulsating: {
-      position: 'absolute',
-      left: 0,
-      top: 0,
-      display: 'block',
-      width: '100%',
-      height: '100%',
-      animation: `mui-ripple-pulsate 1500ms ${theme.transitions.easing.easeInOut} 200ms infinite`,
-      rippleVisible: {
-        opacity: 0.2,
-      },
+export const styles: StyleRulesCallback<TouchRippleClassKey> = (theme: Theme) => ({
```

Comment:
FYI, you don’t need the `: Theme` annotation since you declared the overall type of `styles`.

Reply:
Done thanks.